contracts/MarginBase.sol

```diff
@@ -412,4 +454,148 @@ contract MarginBase is MinimalProxyable {
     {
         return IFuturesMarket(futuresManager.marketForKey(_marketKey));
     }
+
+    /// @notice exchangeRates() fetches current ExchangeRates contract
+    function exchangeRates() internal view returns (IExchangeRates) {
+        return
+            IExchangeRates(
+                addressResolver.requireAndGetAddress(
+                    "ExchangeRates",
+                    "MarginBase: Could not get ExchangeRates"
+                )
+            );
+    }
+
+    /*///////////////////////////////////////////////////////////////
+                            Limit Orders
+    ///////////////////////////////////////////////////////////////*/
+
+    /// @notice limit order logic condition checker
+    /// @param _orderId: key for an active order
+    function validOrder(uint256 _orderId) public view returns (bool) {
+        Order memory order = orders[_orderId];
+
+        bytes32 currencyKey = futuresMarket(order.marketKey).baseAsset();
+        // Get exchange rate for 1 unit
+        uint256 price = exchangeRates().effectiveValue(
+            currencyKey,
+            1e18,
+            "sUSD"
+        );
+
+        if (order.sizeDelta > 0) {
+            // Long
+            return price <= order.desiredPrice;
+        } else if (order.sizeDelta < 0) {
+            // Short
+            return price >= order.desiredPrice;
+        } else {
+            revert("Order size 0");
```

Comment:
Would `order.sizeDelta` ever be 0?

Reply:
it can be, but i'll move this to the placeOrder function that way you can't place a size 0 order